scripts/create-package/plopfile.ts

```diff
@@ -204,20 +190,24 @@ function replaceVersionsFromReference(
       } else if (referenceDeps[depType]?.[depPkg]) {
         newPackageJson[depType][depPkg] = referenceDeps[depType][depPkg];
       } else {
-        hasError = true;
-        console.warn(
-          chalk.yellow(
-            `Couldn't determine appropriate ${depType} version of ${depPkg} from ${referencePackages.join(
-              ' or ',
-            )} package.json`,
-          ),
-        );
+        // Record the error and wait to throw until later for better logs
+        errorPackages.push(`${depPkg} (${depType})`);
       }
     }
   }
-  return hasError;
+  if (errorPackages.length) {
+    throw new Error(
+      `Couldn't determine appropriate version of ${errorPackages.length} packages from ${referencePackages.join(
```

Comment:
perhaps we could stop doing this and set references packages with `*` syncpack should scream on this and prompt the dev to actually set the correct versions